File: vol1/ch6/LeetCode543.cpp
class Solution {
    int ans;                //存放答案
public:
    int diameterOfBinaryTree(TreeNode* root) {
        ans=0;
        depth(root);
        return ans;
    }
    int depth(TreeNode* root) {
        if(root==NULL) return 0;
        int leftd=depth(root->left);
        int rightd=depth(root->right);
        ans=max(ans,leftd+rightd);  //每个结点的等于左子树深度+右子树深度
        return max(leftd,rightd)+1;//返回结点r的深度
    }
};

File: vol1/ch6/LeetCode543.py
class Solution:
    def diameterOfBinaryTree(self, root: TreeNode) -> int:
        self.ans=0
        self.depth(root)
        return self.ans

    def depth(self,root):
        if root==None: return 0
        leftdepth=self.depth(root.left)
        rightdepth=self.depth(root.right)
        self.ans=max(self.ans,leftdepth+rightdepth)
        return max(leftdepth,rightdepth)+1
